apollo-router/src/cache.rs

```diff
@@ -1,231 +0,0 @@
-                    drop(locked_wait_map);
-
-                    let value = {
-                        // when _drop_signal is dropped, either by getting out of the block, returning
-                        // the error from ready_oneshot or by cancellation, the drop_sentinel future will
-                        // return with Err(), then we remove the entry from the wait map
-                        let (_drop_signal, drop_sentinel) = oneshot::channel::<()>();
-                        let wait_map = self.wait_map.clone();
-                        let k = key.clone();
-                        tokio::task::spawn(async move {
-                            let _ = drop_sentinel.await;
-                            let mut locked_wait_map = wait_map.lock().await;
-                            locked_wait_map.remove(&k);
-                        });
-
-                        // This is the potentially high duration operation where we ask our resolver to
-                        // resolve the key (retrieve a value) for us
-                        // No cache locks are held here
-                        let value = self.resolver.retrieve(key.clone()).await;
-
-                        // Update our cache
-                        let mut locked_cache = self.cached.lock().await;
-                        locked_cache.put(key.clone(), value.clone());
-                        value
-                    };
-
-                    // Let our waiters know
-                    let broadcast_value = value.clone();
-                    // We may get errors here, for instance if a task is cancelled,
-                    // so just ignore the result of send
-                    let _ = tokio::task::spawn_blocking(move || {
-                        tx.send((key, broadcast_value))
-                    })
-                    .await
-                    .expect("can only fail if the task is aborted or if the internal code panics, neither is possible here; qed");
-                    return value;
-                }
-            }
-        }
-    }
-}
-
-#[cfg(test)]
-mod tests {
-    use async_trait::async_trait;
-    use futures::stream::FuturesUnordered;
-    use futures::stream::StreamExt;
-    use mockall::mock;
-    use test_log::test;
-
-    use super::*;
-    use crate::error::CacheResolverError;
-
-    struct HasACache {
-        cm: CachingMap<usize, usize>,
-    }
-
-    struct HasACacheResolver {}
-
-    impl HasACache {
-        // fn new(resolver: limit: usize) -> Self {
-        fn new(
-            resolver: Box<(dyn CacheResolver<usize, usize> + Send + Sync)>,
-            cache_limit: usize,
-        ) -> Self {
-            // let resolver = Box::new(HasACacheResolver {});
-            let cm = CachingMap::new(resolver, cache_limit);
-            Self { cm }
-        }
-
-        async fn get(&self, key: usize) -> Result<usize, CacheResolverError> {
-            self.cm.get(key).await
-        }
-    }
-
-    #[async_trait]
```

Comment:
I liked that these tests checked cache limits and delegation were working properly. Can't we preserve them with the re-working?

Reply:
sure

Reply:
fixed in eda738d6